### ISSUE TITLE
Early stopping in LightGbm

### ISSUE THREAD

**Issue by petterton** [dotnet/machinelearning#752]
I am trying to implement a LightGbm classifier with early stopping, but I can not figure how to enter a validation set into the pipeline. I can see that EarlyStoppingRound can be set for LightGbm, so I suppose there should be a way to set a validation set. My current code:

```
var pipeline = new LearningPipeline
{
    new TextLoader(TrainDataPath).CreateFrom<MyFeature>(useHeader: true),
    new ColumnConcatenator("Features", ... ),
    new LightGbmBinaryClassifier
    {
        NumLeaves = 31, 
        NumBoostRound = 100, 
        EarlyStoppingRound = 10,
    },
};
_model = pipeline.Train<MyFeature, MyPrediction>();

```

**Comment by Ivanidzo4ka** [comment]
With current learning Api, it's not possible.

You can use our internal Api like: https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.Tests/Scenarios/Api/TrainWithValidationSet.cs

Both of this API would be deprecated by 1.0 since they either too complicated or don't provide functionality.

**Comment by petterton** [comment]
@Ivanidzo4ka - With the new Api available: Is this possible now? I can see that `EarlyStoppingRound` is now available in the `LightGbmArguments` class. But I can not see a way to set a validation set...

**Comment by Ivanidzo4ka** [comment]
You are correct, we forgot to expose validation set usage.
#1709  should fix that problem.
